data/exporter.go

```diff
@@ -0,0 +1,118 @@
+package data
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/blocknative/dreamboat/beacon"
+	"github.com/blocknative/dreamboat/structs"
+	"github.com/lthibault/log"
+)
+
+type ExportService struct {
+	logger   log.Logger
+	requests chan exportRequest
+	datadir  string
+}
+
+func NewExportService(logger log.Logger, datadir string, bufSize int) ExportService {
+	return ExportService{logger: logger, datadir: datadir, requests: make(chan exportRequest, bufSize)}
+}
+
+func (s ExportService) RunParallel(ctx context.Context, numWorkers int) error {
+	logger := s.logger.WithField("service", "data-exporter")
+
+	datadir := fmt.Sprintf("%s/blockBidAndTrace", s.datadir)
+	if err := os.MkdirAll(datadir, 0755); err != nil {
+		return fmt.Errorf("failed to create datadir: %w", err)
+	}
+
+	for i := 0; i < numWorkers; i++ {
+		files := exportFiles{}
+		// create files
+		// // BlockBidAndTrace
+		filename := fmt.Sprintf("%s/blockBidAndTrace/output_%d.json", s.datadir, i)
+		file, err := os.Create(filename)
+		if err != nil {
+			return fmt.Errorf("failed to create file: %w", err)
+		}
+		files.BlockBidAndTrace = file
```

Comment:
This has to be called inside worker that would take full temporal ownership of the file handler - as I explained. This way you don't deal either with reading and writing from/to the same file by external process

Reply:
If it's dealed by the worker, then we cannot fail on initialization. But we need to propagate the error into `main`, once the workers start running

Reply:
If you like to test if given path is writeable - it's fine to do so on the startup (like touch).
But in this case - we would definitely like to have a size limitted (rotated by relay) files. As this ones would be problematic to deal with after submission